packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/InputTypes/InputType.cs

```diff
@@ -27,20 +29,22 @@ internal InputType GetCollectionEquivalent(InputType inputType)
                     return new InputArrayType(
                         listType.Name,
                         listType.CrossLanguageDefinitionId,
-                        listType.ValueType.GetCollectionEquivalent(inputType));
+                        listType.ValueType.GetCollectionEquivalent(inputType),
+                        listType.Decorators);
                 case InputDictionaryType dictionaryType:
                     return new InputDictionaryType(
                         dictionaryType.Name,
                         dictionaryType.KeyType,
-                        dictionaryType.ValueType.GetCollectionEquivalent(inputType));
+                        dictionaryType.ValueType.GetCollectionEquivalent(inputType),
+                        dictionaryType.Decorators);
                 default:
                     return inputType;
             }
         }
         public InputType WithNullable(bool isNullable)
         {
             if (isNullable)
-                return new InputNullableType(this);
+                return new InputNullableType(this, this.Decorators);
```

Comment:
does it make sense here?

The `this.Decorators` are the decorators of this type (without nullable), the constructed nullable type does not have the decorators. We should not pass it around.

A nullable type might have decorators, but that is something the emitter tells us, not when we construct a nullable type in the C# side.